File: Calculator/app/src/main/java/com/example/calculator/MainActivity.java
package com.example.calculator;

import androidx.appcompat.app.AppCompatActivity;
import androidx.cardview.widget.CardView;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.GridLayout;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    GridLayout mainGridLayout;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mainGridLayout = (GridLayout)findViewById(R.id.mainGridLayout);

        setEventGridClicked(mainGridLayout);
    }

    private void setEventGridClicked(final GridLayout mainGridLayout) {
        for (int i = 0; i<mainGridLayout.getChildCount();i++){
            final CardView cardView = (CardView) mainGridLayout.getChildAt(i);
            final int finalI = i;
            cardView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    if (finalI == 0){
                        Intent intent = new Intent(getApplicationContext(),IncomeAcitivity.class);
                        startActivity(intent);
                    }
                    else if(finalI == 1){
                        Intent intent = new Intent(getApplicationContext(),NoteDetailIncome.class);
                        startActivity(intent);
                    }
                    else if(finalI == 2){

                    }
                    else if(finalI == 3){

                    }
                    else if(finalI == 4){

                    }
                    else if(finalI == 5){

                    }

                }
            });
        }

    }
}

File: Email Clone/app/src/main/java/com/example/intentimplicitexample/MainActivity.kt
package com.example.intentimplicitexample

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity
import com.example.intentimplicitexample.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    lateinit var binding: ActivityMainBinding

    val TAG = "intent"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        var view = binding.root
        setContentView(view)

        binding.btnSend.setOnClickListener {
            Log.d(TAG, "create intent")
            val receiverName = binding.edSendTo.text.toString().split(',').toTypedArray()
            val emailTitle: String = binding.edEmailTitle.text.toString()
            val emailBody: String = binding.edEmailBody.text.toString()
            val sendEmailIntent = Intent().apply {
                action = Intent.ACTION_SEND
                putExtra(Intent.EXTRA_EMAIL, receiverName)
                putExtra(Intent.EXTRA_SUBJECT, emailTitle)
                putExtra(Intent.EXTRA_TEXT, emailBody)
                type = "text/plain"
                Log.d(TAG, "success created intent")
            }

            if (sendEmailIntent.resolveActivity(packageManager) != null) {
                startActivity(sendEmailIntent)
            }
        }
    }
}